### PR TITLE
Coverage/reseller and nobroker role

### LINKED ISSUE #72: reseller-broker and no-broker
This logic:

https://github.com/openactive/OpenActive.Server.NET/blob/29cff7750305c3a45a50f9d32fe70e5f8c419aaf/OpenActive.Server.NET/StoreBookingEngine/StoreBookingEngine.cs#L454-L464

Should be moved to underneath here this logic :

https://github.com/openactive/OpenActive.Server.NET/blob/17bbf241420ddedaba304656a8b52fbff52bb2e5/OpenActive.Server.NET/CustomBookingEngine/CustomBookingEngine.cs#L667-L670

And should also include the following:
- `customer` is optional in `ResellerBroker` mode
- if `broker` is supplied in `NoBroker` mode should throw `IncompleteBrokerDetailsError` (placeholder for https://github.com/openactive/open-booking-api/issues/167)

### PR DESCRIPTION
**Pull request by NemanjaStefanovic**
Added checks for reseller and nobroker role.
I split it up in 2 commits (moving 2 ifs and extending for new checks)

Closes #72

### CODE REVIEW

**Comment by lukehesluke** [comment]
There looks to be one error:

Screenshot 2020-11-05 at 19 24 34

**Comment by civsiv** [comment]
Works locally against https://github.com/openactive/openactive-test-suite/pull/293 (however having issues with CI, so not merged)

File: OpenActive.Server.NET/CustomBookingEngine/CustomBookingEngine.cs [hunk 1]
```diff
@@ -669,16 +674,38 @@ protected BookingFlowContext ValidateFlowRequest<TOrder>(OrderIdComponents order
...
 
+            if (order.BrokerRole == BrokerType.NoBroker && order.Broker != null)
+            {
+                throw new OpenBookingException(new IncompleteBrokerDetailsError()); // TODO: Placeholder for https://github.com/openactive/open-booking-api/issues/167
+            }
+
+            // Throw error on incomplete customer details if C2, P or B
+            if (stage != FlowStage.C1 && (order.Customer == null || string.IsNullOrWhiteSpace(order.Customer.Email)))
+            {
+                // Unless it's B when the Broker isn't a ResellerBroker
```
Comment (nickevansuk) [inline OpenActive.Server.NET/CustomBookingEngine/CustomBookingEngine.cs:685]:
```suggestion
                // Unless it's B, P or C2 when the Broker isn't a ResellerBroker
```

**Review by nickevansuk — CHANGES_REQUESTED** [review]
One minor suggestion

**Pull request merged**